src/libraries/System.Net.Http/src/System/Net/Http/WasiHttpHandler/WasiInputStream.cs

```diff
@@ -158,6 +167,56 @@ public override async ValueTask<int> ReadAsync(
             return result;
         }
 
+        private async Task ReadTrailingHeaders(CancellationToken cancellationToken)
+        {
+            isClosed = true;
+            stream.Dispose();
+            using var futureTrailers = IncomingBody.Finish(body);
+            while (true)
+            {
+                var trailers = futureTrailers.Get();
+                if (trailers is null)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await WasiHttpInterop
+                        .RegisterWasiPollable(futureTrailers.Subscribe(), cancellationToken)
+                        .ConfigureAwait(false);
+                }
+                else
+                {
+                    var inner = ((Result<Result<Fields?, ErrorCode>, None>)trailers!).AsOk;
+                    if (inner.IsOk)
+                    {
+                        using var headers = inner.AsOk;
+                        if (headers is not null)
+                        {
+                            response.StoreReceivedTrailingHeaders(
+                                WasiHttpInterop.ConvertTrailingResponseHeaders(headers)
+                            );
+                        }
+
+                        break;
+                    }
+                    else if (inner.AsErr.Tag == ErrorCode.CONNECTION_TERMINATED)
+                    {
+                        // TODO: As of this writing, `wasmtime-wasi-http`
```

Comment:
FYI, this is fixed in Wasmtime's `main` branch: https://github.com/bytecodealliance/wasmtime/pull/9208

Reply:
should we update the comment ... or add version check?

Reply:
I will update the comment.

BTW: I expect this code evolve in near future. We don't have most of the HTTP functional tests yet, because we don't have sockets yet. Once we do, I'm sure we will discover lot of other small details to be fixed.